proposals/0001-lhls.md

```diff
@@ -0,0 +1,313 @@
+
+HTTP uses TCP, and TCP has inherent latency which is makes it less well suited for streaming data compared to WebRTC. TCP has mechanisms such as flow control, retransmission, and error-free data transfer which are useful for serving sites but are often unnecessary for delivering video. These transfer mechanism each have a latency associated with them which repeats for each request. LHLS mitigates some of this latency of TCP by initiating the prefetch request before the segment is ready.
+
+#### WebRTC
+
+WebRTC relies on UDP for data delivery. UDP achieves significantly lower latency than TCP because it is lossy - it does not spend time guaranteeing the transmission of data. Lost data in a media stream is usually not fatal and results in dropped frames. WebRTC has been shown to achieve sub-second streaming latency.
+
+WebRTC suffers from scalability. Each "peer", in this case the client, requires a direct connection to another peer. Streaming servers are able to act as peers but must serve each request with a unique connection. This requirement precludes edge caching. A fitting analogy is that WebRTC streams require a unique URL for every client. WebRTC does have the advantage that peers may also be clients but this requires additional code.
+
+WebRTC also is less accessible than HTTP. Most developers today do not stream via WebRTC, and may not have the knowledge or resources to do so. Client support for WebRTC is available but more limited than existing HTTP clients. SSAI solutions are possible but also require further changes, and the breadth of support is not clear.
+
+## Method of Advertising LHLS Segments
+
+#### In-Playlist Signaling
+
+In-playlist signaling happens when prefetch segments are advertised within the same playlist as complete segments. Prefetch segments are included at the bottom and are differentiated via the `EXT-X-PREFETCH` tag. This is the scheme outlined in this proposal. 
+
+We chose this scheme because it is backwards compatible with clients which have not implemented this proposal. According to the HLS specification clients must ignore any tags not outlined within; since `EXT-X-PREFETCH` is not an official tag clients will be left with a standard HLS playlist. This allows a server to deliver a single manifest which plays in both LHLS and unsupported clients. This scheme also allows clients to disable or enable LHLS while the client is streaming. Developers may make this option available to the user so that they are able to choose an experience most beneficial to them.
```

Comment:
```suggestion
We chose this scheme because it is backwards compatible with clients which have not implemented this proposal. According to the HLS specification, clients must ignore any tags not outlined within; since `EXT-X-PREFETCH` is not an official tag, clients will be left with a standard HLS playlist. This allows a server to deliver a single manifest which plays in both LHLS clients and HLS clients. This scheme also allows clients to disable or enable LHLS while the client is streaming. Developers may make this option available to the user so that they are able to choose the experience that is most beneficial to them.
```